content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,423 @@
+
+Once you have your environment set up, you can install the dependencies by
+running:
+
+```bash
+pip install -r requirements.txt
+```
+
+This will download all of the packages you need to run the example. Now you have
+everything you need to get started with experiments and checkpoints.
+
+## Setting up a DVC pipeline
+
+If you don't have a DVC pipeline setup in your project, adding one only takes a
+few commands. At the root of the project, run:
+
+```bash
+dvc init
+```
+
+This sets up the files you need for your DVC pipeline to work.
+
+This will create a _dvc.yaml_ file in your root directory.
+
+Now we need to add a stage for training our model within a DVC pipeliene. We'll
+do that with the following command:
+
+```bash
+dvc stage add -n train -d data/MNIST -d train.py -c model.pt --plots-no-cache predictions.json -p seed,lr,weight_decay --live dvclive python train.py
+```
+
+Before we go any further, this is a great point to add these changes to your Git
+history. You can do that with the following commands:
+
+```bash
+git add .
+git commit -m "created DVC pipeline"
+```
+
+## Enabling DVC pipelines for checkpoint experiments
+
+After running the command above to setup your _train_ stage, your _dvc.yaml_
+should have the following code.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory. This will create a new stage with checkpoints enabled, but
+does not update existing stages.
+
+The rest of the command `dvc stage add` sets up our dependencies for running the
+training code, which parameters we want to track (which are defined in the
+_params.yaml_), some configurations for our plots and showing the training
+metrics.
+
+We'll manually add a slight modification to the way our plots are handled. Add
+the following lines to your _dvc.yaml_. This just changes the way our results
+are shown after we've run some experiments.
+
+```yaml
+plots:
+  - predictions.json:
+      cache: false
+      template: confusion
+      x: actual
+      y: predicted
+```
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
```

Comment:
A code snippet might be helpful:
```python
        # Iterate over training epochs.
        for i in range(1, EPOCHS+1):
            # Train in batches.
            train_loader = torch.utils.data.DataLoader(
                    dataset=list(zip(x_train, y_train)),
                    batch_size=512,
                    shuffle=True)
            for x_batch, y_batch in train_loader:
                train(model, x_batch, y_batch, params["lr"], params["weight_decay"])
            torch.save(model.state_dict(), "model.pt")
            # Evaluate and checkpoint.
            metrics = evaluate(model, x_test, y_test)
            for k, v in metrics.items():
                print('Epoch %s: %s=%s'%(i, k, v))
```

Reply:
Also, you can mention that the line `torch.save(model.state_dict(), "model.pt")` updates the checkpoint file.